### PR TITLE
BUG: Retain tz transformation in groupby.transform

### LINKED ISSUE #27496: Lost timezone information after groupby transform
#### Code Sample, a copy-pastable example if possible

Similar issue as #24198 in pandas 0.25.0

```python
import pandas as pd

df = pd.DataFrame({'time': [pd.Timestamp('2010-07-15 03:14:45'), 
                            pd.Timestamp('2010-11-19 18:47:06')],
                   'timezone': ['Etc/GMT+4', 'US/Eastern']})

df['time_tz'] = (df.groupby(['timezone'])['time']
                   .transform(lambda x: x.dt.tz_localize(x.name, ambiguous='NaT')))

df['time_tz_desired'] = (df.groupby(['timezone'])['time']
                   .apply(lambda x: x.dt.tz_localize(x.name, ambiguous='NaT')))
print(df)
```
#### Problem description

The timezone in the transform should be preserved but it is not. 
For the apply the datetime returned is an object type

`df.info()` shows
```
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 4 columns):
time               2 non-null datetime64[ns]
timezone           2 non-null object
time_tz            2 non-null datetime64[ns]
time_tz_desired    2 non-null object
dtypes: datetime64[ns](2), object(2)
memory usage: 192.0+ bytes
```

#### Expected Output
Should preserve timezone
```python
df.groupby(['timezone'])['time'].transform(lambda x: x.dt.tz_localize(x.name, ambiguous='NaT'))
```
And should return datetime
```python
df.groupby(['timezone'])['time'].apply(lambda x: x.dt.tz_localize(x.name, ambiguous='NaT'))
```

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.3.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 0.25.0
numpy            : 1.16.3
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 40.8.0
Cython           : 0.29.10
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.3.4
html5lib         : 1.0.1
pymysql          : 0.9.3
psycopg2         : 2.8.2 (dt dec pq3 ext lo64)
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: None
bs4              : 4.7.1
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.3.4
matplotlib       : 3.0.3
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.3.0
sqlalchemy       : 1.3.3
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #27496
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1025,8 +1025,8 @@ def transform(self, func, *args, **kwargs):
             object.__setattr__(group, "name", name)
             res = wrapper(group)
 
-            if hasattr(res, "values"):
-                res = res.values
+            if hasattr(res, "_values"):
```
Comment (jreback) [inline pandas/core/groupby/generic.py]: could do

res = getattr(res, ‘_values’, res)

Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: I think Joris opened an issue about this recently; can we make this a more explicit isinstace ABCWhatever check?

Comment (WillAyd) [reply pandas/core/groupby/generic.py]: I think this is ref #27167 so isinstance seems to be the direction rather than getattr

**Review by WillAyd — APPROVED** [review]
lgtm

**Review by TomAugspurger — APPROVED** [review]

**Review by TomAugspurger — APPROVED** [review]

**Pull request merged**